ledger/src/leader_schedule_cache.rs

```diff
@@ -48,7 +48,9 @@ impl LeaderScheduleCache {
 
         // Calculate the schedule for all epochs between 0 and leader_schedule_epoch(root)
         let leader_schedule_epoch = epoch_schedule.get_leader_schedule_epoch(root_bank.slot());
-        for epoch in 0..leader_schedule_epoch {
+        for epoch in (leader_schedule_epoch.max(MAX_LEADER_SCHEDULE_STAKES)
```

Comment:
this might come out prettier if the bank gave you the epochs for which it had leader_schedule_stakes...